pkg/kritis/cryptolib/authenticated_attestation_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+import "testing"
+
+const validPayload = `{
+    "critical": {
+        "identity": {
+            "docker-reference": "gcr.io/google-samples/hello-app"
+        },
+        "image": {
+            "docker-manifest-digest": "sha256:bedb3feb23e81d162e33976fd7b245adff00379f4755c0213e84405e5b1e0988"
+        },
+    "type": "Google cloud binauthz container signature"
+    }
+}`
+
+const invalidPayload = `{
+    "invalid-field": invalid
+}`
+
+func TestFormAuthenticatedAttestation(t *testing.T) {
+	tcs := []struct {
+		name        string
+		payload     []byte
+		expectedErr bool
+		expected    authenticatedAttestation
+	}{
+		{
+			name:        "correct authenticated attestation",
+			payload:     []byte(validPayload),
+			expectedErr: false,
+			expected: authenticatedAttestation{
+				ImageName:   "gcr.io/google-samples/hello-app",
+				ImageDigest: "sha256:bedb3feb23e81d162e33976fd7b245adff00379f4755c0213e84405e5b1e0988",
+			},
+		},
+		{
+			name:        "cannot unmarshal payload",
+			payload:     []byte(invalidPayload),
+			expectedErr: true,
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, err := formAuthenticatedAttestation(tc.payload)
+			if tc.expectedErr != (err != nil) {
+				t.Fatalf("formAuthenticatedAttestation(_) got %v, wanted error? = %v", err, tc.expectedErr)
```

Comment:
no question mark after wanted error

Nit: the standard formatting is: YourFunc(%v) = %v, want %v

go/go-test-comments#got-before-want

Reply:
Done.